### ISSUE TITLE
Could not find CMake version 3.28.3-msvc11

### ISSUE THREAD

**Issue by dofuuz** [scikit-build/scikit-build-core#833]
I am trying to use MSVC bundled CMake. But scikit_build_core couldn't find it.
```
d:\backup10\work\python-soxr>pip install --no-build-isolation -ve .
Using pip 24.0 from C:\dev\WPy64-31241\python-3.12.4.amd64\Lib\site-packages\pip (python 3.12)
Obtaining file:///D:/backup10/work/python-soxr
  Running command Checking if build backend supports build_editable
  Checking if build backend supports build_editable ... done
  Running command Preparing editable metadata (pyproject.toml)
  2024-07-27 22:59:27,275 - scikit_build_core - WARNING - Could not determine CMake version, got '{"debugger":true,"fileApi":{"requests":[{"kind":"codemodel","version":[{"major":2,"minor":6}]},{"kind":"configureLog","version":[{"major":1,"minor":0}]},{"kind":"cache","version":[{"major":2,"minor":0}]},{"kind":"cmakeFiles","version":[{"major":1,"minor":0}]},{"kind":"toolchains","version":[{"major":1,"minor":0}]}]},"generators":[{"extraGenerators":["Kate"],"name":"Ninja Multi-Config","platformSupport":false,"toolsetSupport":false},{"extraGenerators":[],"name":"Visual Studio 17 2022","platformSupport":true,"supportedPlatforms":["x64","Win32","ARM","ARM64","ARM64EC"],"toolsetSupport":true},{"extraGenerators":[],"name":"Visual Studio 14 2015 ARM","platformSupport":true,"supportedPlatforms":["x64","Win32","ARM"],"toolsetSupport":true},{"extraGenerators":[],"name":"Visual Studio 16 2019","platformSupport":true,"supportedPlatforms":["x64","Win32","ARM","ARM64","ARM64EC"],"toolsetSupport":true},{"extraGenerators":[],"name":"Visual Studio 15 2017 Win64","platformSupport":true,"supportedPlatforms":["x64","Win32","ARM","ARM64"],"toolsetSupport":true},{"extraGenerators":[],"name":"Visual Studio 15 2017","platformSupport":true,"supportedPlatforms":["x64","Win32","ARM","ARM64"],"toolsetSupport":true},{"extraGenerators":[],"name":"Visual Studio 15 2017 ARM","platformSupport":true,"supportedPlatforms":["x64","Win32","ARM","ARM64"],"toolsetSupport":true},{"extraGenerators":[],"name":"Visual Studio 12 2013","platformSupport":true,"supportedPlatforms":["x64","Win32","ARM"],"toolsetSupport":true},{"extraGenerators":[],"name":"Visual Studio 14 2015 Win64","platformSupport":true,"supportedPlatforms":["x64","Win32","ARM"],"toolsetSupport":true},{"extraGenerators":[],"name":"Visual Studio 12 2013 Win64","platformSupport":true,"supportedPlatforms":["x64","Win32","ARM"],"toolsetSupport":true},{"extraGenerators":[],"name":"Visual Studio 14 2015","platformSupport":true,"supportedPlatforms":["x64","Win32","ARM"],"toolsetSupport":true},{"extraGenerators":["CodeBlocks","CodeLite","Eclipse CDT4","Kate","Sublime Text 2"],"name":"MinGW Makefiles","platformSupport":false,"toolsetSupport":false},{"extraGenerators":[],"name":"Visual Studio 12 2013 ARM","platformSupport":true,"supportedPlatforms":["x64","Win32","ARM"],"toolsetSupport":true},{"extraGenerators":[],"name":"Visual Studio 9 2008","platformSupport":true,"supportedPlatforms":["x64","Win32","Itanium"],"toolsetSupport":false},{"extraGenerators":[],"name":"Visual Studio 9 2008 Win64","platformSupport":true,"supportedPlatforms":["x64","Win32","Itanium"],"toolsetSupport":false},{"extraGenerators":["CodeBlocks"],"name":"NMake Makefiles JOM","platformSupport":false,"toolsetSupport":false},{"extraGenerators":[],"name":"Visual Studio 9 2008 IA64","platformSupport":true,"supportedPlatforms":["x64","Win32","Itanium"],"toolsetSupport":false},{"extraGenerators":[],"name":"MSYS Makefiles","platformSupport":false,"toolsetSupport":false},{"extraGenerators":[],"name":"Borland Makefiles","platformSupport":false,"toolsetSupport":false},{"extraGenerators":["CodeBlocks","CodeLite","Eclipse CDT4","Kate","Sublime Text 2"],"name":"NMake Makefiles","platformSupport":false,"toolsetSupport":false},{"extraGenerators":[],"name":"Green Hills MULTI","platformSupport":true,"toolsetSupport":true},{"extraGenerators":["CodeBlocks","CodeLite","Eclipse CDT4","Kate","Sublime Text 2"],"name":"Unix Makefiles","platformSupport":false,"toolsetSupport":false},{"extraGenerators":["CodeBlocks","CodeLite","Eclipse CDT4","Kate","Sublime Text 2"],"name":"Ninja","platformSupport":false,"toolsetSupport":false},{"extraGenerators":[],"name":"Watcom WMake","platformSupport":false,"toolsetSupport":false}],"serverMode":false,"tls":true,"version":{"isDirty":false,"major":3,"minor":28,"patch":3,"string":"3.28.3-msvc11","suffix":"msvc11"}}\n'
  Traceback (most recent call last):
    File "C:\dev\WPy64-31241\python-3.12.4.amd64\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
      main()
    File "C:\dev\WPy64-31241\python-3.12.4.amd64\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\dev\WPy64-31241\python-3.12.4.amd64\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 181, in prepare_metadata_for_build_editable
      return hook(metadata_directory, config_settings)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\dev\WPy64-31241\python-3.12.4.amd64\Lib\site-packages\scikit_build_core\build\__init__.py", line 85, in prepare_metadata_for_build_editable
      return _build_wheel_impl(
             ^^^^^^^^^^^^^^^^^^
    File "C:\dev\WPy64-31241\python-3.12.4.amd64\Lib\site-packages\scikit_build_core\build\wheel.py", line 151, in _build_wheel_impl
      cmake = CMake.default_search(version=settings.cmake.version, env=os.environ)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\dev\WPy64-31241\python-3.12.4.amd64\Lib\site-packages\scikit_build_core\cmake.py", line 63, in default_search
      raise CMakeNotFoundError(msg)
  scikit_build_core.errors.CMakeNotFoundError: Could not find CMake with version >=3.15
  error: subprocess-exited-with-error

  × Preparing editable metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> See above for output.

  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: 'C:\dev\WPy64-31241\python-3.12.4.amd64\python.exe' 'C:\dev\WPy64-31241\python-3.12.4.amd64\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py' prepare_metadata_for_build_editable 'C:\Users\dof\AppData\Local\Temp\tmp1k62spn7'
  cwd: D:\backup10\work\python-soxr
  Preparing editable metadata (pyproject.toml) ... error
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

[notice] A new release of pip is available: 24.0 -> 24.1.2
[notice] To update, run: python.exe -m pip install --upgrade pip
```

The bundled CMake version is `3.28.3-msvc11`.
```
d:\backup10\work\python-soxr>cmake --version
cmake version 3.28.3-msvc11

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

Possible fix:
https://github.com/scikit-build/scikit-build-core/blob/9771da41dde5d992294754e5a8e6def2e4f33bcf/src/scikit_build_core/program_search.py#L83

Changing it like [line 93](https://github.com/scikit-build/scikit-build-core/blob/9771da41dde5d992294754e5a8e6def2e4f33bcf/src/scikit_build_core/program_search.py#L93) seems like works.
```python
            version = Version(json.loads(result.stdout)["version"]["string"].split("-")[0])
```

**Comment by henryiii** [comment]
I thought we fixed this… Maybe it regressed when we moved to cmake -E?

**Comment by henryiii** [comment]
Yes, it was only implemented with `--version`: https://github.com/scikit-build/scikit-build-core/blob/9771da41dde5d992294754e5a8e6def2e4f33bcf/src/scikit_build_core/program_search.py#L93

**Comment by henryiii** [comment]
Released 0.9.9 with this fix.

**Comment by dofuuz** [comment]
Thank you for fast fix 👍
